src/Stratis.Bitcoin.IntegrationTests/Mempool/MempoolRelaySpecification_Steps.cs

```diff
@@ -0,0 +1,97 @@
+
+            this.nodeBuilder.StartAll();
+            this.nodeA.NotInIBD();
+            this.nodeB.NotInIBD();
+            this.nodeC.NotInIBD();
+
+            this.coinbaseMaturity = (int)this.nodeA.FullNode.Network.Consensus.Option<PowConsensusOptions>().CoinbaseMaturity;
+
+            // add some coins to nodeA
+            this.nodeA.SetDummyMinerSecret(new BitcoinSecret(new Key(), this.nodeA.FullNode.Network));
+            this.nodeA.GenerateStratisWithMiner(this.coinbaseMaturity + 1);
+            TestHelper.WaitLoop(() => TestHelper.IsNodeSynced(this.nodeA));
+        }
+
+        protected void nodeA_connects_to_nodeB()
+        {
+            this.nodeA.CreateRPCClient().AddNode(this.nodeB.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.nodeA, this.nodeB));
+        }
+
+        protected void nodeA_nodeB_and_nodeC_are_not_whitelisted()
+        {
+            this.nodeA.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+            this.nodeB.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+            this.nodeC.FullNode.NodeService<IConnectionManager>().ConnectedPeers.First().Behavior<ConnectionManagerBehavior>().Whitelisted = false;
+        }
+
+        protected void nodeB_connects_to_nodeC()
+        {
+            this.nodeB.CreateRPCClient().AddNode(this.nodeC.Endpoint, true);
+            TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(this.nodeA, this.nodeB));
+        }
+
+        protected void nodeA_creates_a_transaction_and_propagates_to_nodeB()
+        {
+            var block = this.nodeA.FullNode.BlockStoreManager().BlockRepository.GetAsync(this.nodeA.FullNode.Chain.GetBlock(1).HashBlock).Result;
+            var prevTrx = block.Transactions.First();
+            var dest = new BitcoinSecret(new Key(), this.nodeA.FullNode.Network);
+
+            this.transaction = new Transaction();
+            this.transaction.AddInput(new TxIn(new OutPoint(prevTrx.GetHash(), 0), PayToPubkeyHashTemplate.Instance.GenerateScriptPubKey(this.nodeA.MinerSecret.PubKey)));
+            this.transaction.AddOutput(new TxOut("25", dest.PubKey.Hash));
+            this.transaction.AddOutput(new TxOut("24", new Key().PubKey.Hash)); // 1 btc fee
+            this.transaction.Sign(this.nodeA.MinerSecret, new Coin(this.transaction.Inputs.First().PrevOut, prevTrx.Outputs.First()));
+
+            this.nodeA.Broadcast(this.transaction);
+        }
+
+        protected void the_transaction_is_propagated_to_nodeC()
+        {
+            var rpc = this.nodeC.CreateRPCClient();
+            TestHelper.WaitLoop(() => rpc.GetRawMempool().Any());
+            var trxid = this.nodeC.CreateRPCClient().GetRawMempool().Should().ContainSingle().Which;
+            trxid.IsSameOrEqualTo(this.transaction.GetHash());
```

Comment:
ah ok - it is 2 lines but the .Which implied it followed on, can it just be combined?

Reply:
Ah yeah, initially it did not work for me for some reason but for sure this is more correct